src/blocked.c

```diff
@@ -279,323 +270,18 @@ void disconnectAllBlockedClients(void) {
     }
 }
 
-/* Helper function for handleClientsBlockedOnKeys(). This function is called
- * when there may be clients blocked on a list key, and there may be new
- * data to fetch (the key is ready). */
-void serveClientsBlockedOnListKey(robj *o, readyList *rl) {
-    /* Optimization: If no clients are in type BLOCKED_LIST,
-     * we can skip this loop. */
-    if (!server.blocked_clients_by_type[BLOCKED_LIST]) return;
-
-    /* We serve clients in the same order they blocked for
-     * this key, from the first blocked to the last. */
-    dictEntry *de = dictFind(rl->db->blocking_keys,rl->key);
-    if (de) {
-        list *clients = dictGetVal(de);
-        listNode *ln;
-        listIter li;
-        listRewind(clients,&li);
-
-        while((ln = listNext(&li))) {
-            client *receiver = listNodeValue(ln);
-            if (receiver->btype != BLOCKED_LIST) continue;
-
-            int deleted = 0;
-            robj *dstkey = receiver->bpop.target;
-            int wherefrom = receiver->bpop.blockpos.wherefrom;
-            int whereto = receiver->bpop.blockpos.whereto;
-
-            /* Protect receiver->bpop.target, that will be
-             * freed by the next unblockClient()
-             * call. */
-            if (dstkey) incrRefCount(dstkey);
-
-            long long prev_error_replies = server.stat_total_error_replies;
-            client *old_client = server.current_client;
-            server.current_client = receiver;
-            monotime replyTimer;
-            elapsedStart(&replyTimer);
-            serveClientBlockedOnList(receiver, o,
-                                     rl->key, dstkey, rl->db,
-                                     wherefrom, whereto,
-                                     &deleted);
-            updateStatsOnUnblock(receiver, 0, elapsedUs(replyTimer), server.stat_total_error_replies != prev_error_replies);
```

Comment:
now we only count the command reprocessed slowlog and latency, but lost the command "processed" time, the last `c->duration`, do we need add the last duration to the current duration?

Reply:
Thats a valid point. I guess we can accumulate the time we processed the command until we reached the decision to block.

Reply:
Actually, the current implementation only incorrectly handles the latency histogram and slowlog.
IMO before this change the pre-attempt duration was calculated twice for blocking commands since both incremented the `c->lastcmd->microseconds` (am I right?) but I guess I should better align this in this PR.

Reply:
@soloestoy - I have made some fixes to statistics. I personally do not like to log the blocking command duration to the command stats before it was unblocked (with the duration) but I did not want to change the existing behavior.
Take a look and tell me what you think.